new-log-viewer: Add tabbed Sidebar with resizable panel; Rearrange open and settings buttons.

junhaoliao opened the pull request:
# References
<!-- Any issues or pull requests relevant to this pull request -->
new-log-viewer series: #45 #46 #48 #51 #52 #53 #54 #55 #56 #59 #60 #61 #62 #63 #66 #67 #68 #69 #70 #71 #72 #73

# Description
<!-- Describe what this request will change/fix and provide any details 
necessary for reviewers -->
1. Add `<SidebarContainer/>`.
2. Add `<FileInfoTab/>`.
3. Move Settings modal toggle button from menu bar to the sidebar.
4. Move File Open button from the menu bar to the side bar.
5. Style fixes in the menu bar and the status bar.

# Validation performed
<!-- What tests and validation you performed on the change -->
1. Referred to https://github.com/y-scope/yscope-log-viewer/pull/46 , started the debug server and loaded a sample log file.
2. Observed the File Info tab getting opened in the left panel by default.
3. Hovered mouse over the right edge of the left panel and the resize handle was highlighted.
4. Dragged the handle:
   1. The panel resized as the handle was dragged, with the right edge of the panel following the mouse cursor.
   2. When dragging towards the left, if the width of the panel was less than a certain size, the panel closed and only the sidebar component is shown.
   3. When dragging towards the right, the panel width will not exceed a certain size, allowing the Editor to be visible at all times.
5. Verified File Open by clicking the button in the top of the sidebar.
6. Verified Settings modal opening by click the settings button in the bottom of the sidebar.
![image](https://github.com/user-attachments/assets/305d4af5-e08f-4644-8d12-24c26110a227)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Introduced a new `FileInfoTabPanel` component to display file information, including file name and size, in a user-friendly format.
	- Added a `CustomListItem` component for improved list item presentation with icons and contextual text.
	- Added a `CentralContainer` component that integrates the `Sidebar`, `DropFileContainer`, and `Editor` for a cohesive layout.
	- Implemented a `ResizeHandle` component for user-friendly resizing of interface elements.

- **Styling Enhancements**
	- Enhanced layout and styling for the new components using Material-UI Joy components and custom CSS.
	- Implemented a responsive grid layout for the `CentralContainer` to optimize space usage.
	- Updated the `StatusBar` styling with a new `z-index` and shadow effect for improved visual layering.

- **Context and State Management**
	- Added new properties to track the original file size and on-disk file size in bytes, improving file handling and display capabilities.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

junhaoliao committed:
Remove redundant div wrapper in Layout component to simplify the DOM structure.

junhaoliao committed:
Add SidebarContainer with resizable tabs.

junhaoliao committed:
Add newline to EOF.

junhaoliao committed:
Add newline to EOF.

junhaoliao committed:
Add borders to StatusBar and MenuBar

junhaoliao committed:
Set resize handle z-index with global CSS variable.

junhaoliao committed:
Introduce 8px padding to .menu-bar for better spacing. Add .menu-bar-filename class to CSS.

junhaoliao commented on new-log-viewer/src/components/MenuBar/index.css:
```diff
@@ -1,6 +1,14 @@
+.menu-bar-filename {
+    align-items: center;
+    flex-grow: 1;
+}
```
```suggestion
}

```

junhaoliao committed:
Add newline to EOF.

junhaoliao committed:
Extract <CustomTabPanel/>.

junhaoliao committed:
Remove redundant line.

kirkrodrigues reviewed (changes requested):
* I find it a bit unintuitive that clicking the file with arrow icon opens an open dialog. I think two things would help:
  1. Replace the icon with a folder icon. Pretty sure that's still the most common icon for opening a file, right? At least Office applications still use it.
  2. Move the button to replace the current file icon in the top left. I think the file icon in the top left is not doing much, especially since we now have the file info as a tab in the sidebar.
     * Alternatively, we could move the open icon above the settings icon, but I think that placement is less intuitive.
* When no file is open, can we hide the icons and perhaps display something like "No file is open."
  * I know in a future PR, we'll probably hide this UI entirely when no file is open, but I think it's a small change to make for now?
* Since "FILE INFO" is the title, the other "File Info" should probably be "Name"?
* Can we add the compressed size and compression ratio to the file info?
* Can we un-highlight the last opened tab when the sidebar is closed? I could see someone misinterpreting it as a notification instead of the last tab that they had open.
* Can we add tool tips for the icons? Perhaps:
  * "Open"
  * "File info"
  * "Settings"
* Do we want to increase the min-width before the sidebar snaps closed?
* This is minor, but can we make the buttons square rather than rectangles?
* Can we decrease the padding in the sidebar by maybe 25%?

junhaoliao committed:
Rename 'File Info' field to 'Name' in FileInfoTab.

junhaoliao committed:
Replace FileOpenIcon with FolderOpenIcon in PanelTabs.

junhaoliao committed:
Move open file functionality to MenuBar.

junhaoliao committed:
Set default messages for no file open.

junhaoliao committed:
Refactor tab panel close behaviour.

junhaoliao committed:
Add TAB_DISPLAY_NAMES and use it for tab titles.

junhaoliao committed:
Refactor panel resize behaviour.

junhaoliao committed:
Merge remote-tracking branch 'yscope/main' into sidebar

# Conflicts:
#	new-log-viewer/src/components/MenuBar/index.css
#	new-log-viewer/src/components/StatusBar/index.css

junhaoliao committed:
Increase the min-width before the sidebar snaps closed.

junhaoliao committed:
Decrease the padding in the sidebar by  25%.

junhaoliao committed:
Make the buttons square rather than rectangles.

junhaoliao committed:
Add YScope logo component and ABeeZee font.

junhaoliao committed:
Add YScopeLogo and replace IconButton with Button.

junhaoliao committed:
Update sidebar tab button dimensions.

junhaoliao commented:
> * I find it a bit unintuitive that clicking the file with arrow icon opens an open dialog. I think two things would help:
>   
>   1. Replace the icon with a folder icon. Pretty sure that's still the most common icon for opening a file, right? At least Office applications still use it.
>   2. Move the button to replace the current file icon in the top left. I think the file icon in the top left is not doing much, especially since we now have the file info as a tab in the sidebar.
>      
>      * Alternatively, we could move the open icon above the settings icon, but I think that placement is less intuitive.

Moved the Open button back into the menu bar.

> * When no file is open, can we hide the icons and perhaps display something like "No file is open."
>   
>   * I know in a future PR, we'll probably hide this UI entirely when no file is open, but I think it's a small change to make for now?

The initial `fileName` and also `logData` are now changed to "No file is open.".

> * Since "FILE INFO" is the title, the other "File Info" should probably be "Name"?

Fixed the typo.

> * Can we add the compressed size and compression ratio to the file info?

Can we add that in another PR? Estimating uncompressed (formatted text) sizes is possible but the calculations for different file types are worth discussions.

> * Can we un-highlight the last opened tab when the sidebar is closed? I could see someone misinterpreting it as a notification instead of the last tab that they had open.

Done.

> * Can we add tool tips for the icons? Perhaps:
>   
>   * "Open"
>   * "File info"
>   * "Settings"

Done by utilizing the "title" prop. In a future PR (soon), we will add JoyUI tooltips.

> * Do we want to increase the min-width before the sidebar snaps closed?

Increased to 250px.

> * This is minor, but can we make the buttons square rather than rectangles?

Done with width and height being 64px. To avoid misalignments with the file open button in the menu bar, added a YScope logo to match the width of the tab button.

> * Can we decrease the padding in the sidebar by maybe 25%?

Done.

junhaoliao committed:
Adjust default Sidebar panel width to 360px.

junhaoliao committed:
Reformat CSS file.

junhaoliao committed:
Remove redundant line in StateContextProvider.tsx

junhaoliao committed:
Disable custom-property-pattern check for joyui custom properties in CSS files.

junhaoliao committed:
Rename 'File Info' tab display name to 'File info'.

junhaoliao committed:
Remove YScopeLogo component and update icon styles.

junhaoliao committed:
Move active tab button indicator placement to the left.

junhaoliao committed:
Change default fileName back to an empty string.

junhaoliao committed:
Show no-file-open message in FileInfoTab.

junhaoliao committed:
Refactor tab buttons to use TooltipTab component which moves indicator to left and provides a tooltip.

junhaoliao committed:
Add arrow to "Open file" tooltip.

junhaoliao committed:
Support clicking on an active tab button to close the tab.

junhaoliao committed:
Remove unnecessary type assertion on tabName value.

junhaoliao committed:
Add set status bar z-index so that the border of it can be shown over the monaco-editor.minimap.

junhaoliao committed:
Change fileName empty string check.

junhaoliao committed:
Add padding between sidebar tab panel title and content.

junhaoliao committed:
Add Divider component to TooltipTab.

junhaoliao committed:
Add outlined variant to TooltipTab component.

junhaoliao committed:
Revert "Add Divider component to TooltipTab."

junhaoliao committed:
Refactor CSS imports.

junhaoliao committed:
Prevent text selection in panel titles.

junhaoliao committed:
Clear original file size before loading.

junhaoliao committed:
Add word-break style to FileInfoTab content.

davemarco reviewed (commented):
I went over the code on and I think it looks good. I didn't test functionality with much rigor. Feel free to come back to harder comments on a later PR if time is an issue

junhaoliao committed:
Merge branch 'main' into sidebar

# Conflicts:
#	new-log-viewer/src/components/MenuBar/index.tsx
#	new-log-viewer/src/contexts/StateContextProvider.tsx

junhaoliao commented on new-log-viewer/src/components/MenuBar/index.tsx:
```diff
@@ -32,54 +26,48 @@ import "./index.css";
+                    title={"Open file"}
+                    variant={"outlined"}
+                >
+                    <IconButton
```
It's to prepare for the search tab panel, which is to be submitted in another PR.

junhaoliao commented on new-log-viewer/src/components/CustomListItem.tsx:
```diff
@@ -0,0 +1,44 @@
+import React from "react";
```
Sure. I was thinking this is reusable across all components but since we don't have the need yet, I agree moving it into the directory where it's referenced (the `SidebarContainer/PanelTabs` directory) would cause less confusions.

junhaoliao committed:
Move CustomListItem.tsx to PanelTabs directory.

junhaoliao commented on new-log-viewer/src/components/Layout.tsx:
```diff
@@ -21,13 +22,13 @@ const Layout = () => {
-                <MenuBar/>
+            <MenuBar/>
+            <SidebarContainer>
                 <DropFileContainer>
                     <Editor/>
                 </DropFileContainer>
-                <StatusBar/>
-            </div>
+            </SidebarContainer>
+            <StatusBar/>
```
I can see where the confusions come from.

I think simply renaming the `<SidebarContainer/>` component can help. Shall we name it `CentralContainer` to indicate the position of it in the <Layout/>? With that said, I think `MainContainer` is meaningful enough.

junhaoliao replied:
Sounds good.

> add a Central container component

done

> with the sidebarContainer(rename sideBar) and the editor as children

done

junhaoliao committed:
Remove unused ABeeZee font link from index.html.

junhaoliao commented on new-log-viewer/src/components/MenuBar/index.css:
```diff
@@ -1,6 +1,28 @@
+}
+
+.menu-bar-open-file-icon {
+    font-size: 24px !important;
```
I never knew the option existed for MUI lol. I tried looking for the equivalent in JoyUI but didn't find anything. Still I agree this is something worth looking into in the near future.

junhaoliao committed:
Fix typo in docs - apply sugguestions.

Co-authored-by: davemarco <83603688+davemarco@users.noreply.github.com>

junhaoliao committed:
Moved custom tab panel title styles from inline CSS to the stylesheet.

junhaoliao commented on new-log-viewer/src/components/SidebarContainer/index.tsx:
```diff
@@ -0,0 +1,101 @@
+        }
+
+        if (activeTabName === tabName) {
+            hidePanelAndResizeHandle();
```
Not sure if I get your ideas in this but I did find the `panelWidth` state a bit unnecessary in the latest code. Note the only meaningful usage of `panelWidth` is in the `useEffect` hook, and in the useEffect hook we're not interacting with any other React states, which is an abusive usage of the `useEffect` hook. 

To rectify that, I removed the `panelWidth` state and now rely on helper functions to get the value from the body style properties.

junhaoliao committed:
Remove useEffect for updating panel width state and externalized width utility functions.

junhaoliao committed:
Remove unnecessary fragment in ResizeHandle component.

junhaoliao committed:
Update variable names and docs for the resize handle behaviours.

junhaoliao commented on new-log-viewer/src/components/SidebarContainer/index.tsx:
```diff
@@ -0,0 +1,101 @@
+        }
+    }, [panelWidth]);
+
+    const handleResize = useCallback((offset: number) => {
```
Let's see if the updated variable names and docs are now more clear.

junhaoliao commented on new-log-viewer/src/components/SidebarContainer/index.tsx:
```diff
@@ -0,0 +1,101 @@
+            <div className={"sidebar-tabs-container"}>
+                <PanelTabs
+                    activeTabName={activeTabName}
+                    ref={tabListRef}
```
PanelTabs is wrapped with React.forwardRef and the ref is passed to an appropriate child element.

junhaoliao replied:
(〃´∀｀)

junhaoliao committed:
Update the `deactivateTabAndHideResizeHandle` function to use the `useCallback` hook.

junhaoliao committed:
Update dependencies for useCallback hooks in SidebarContainer.

junhaoliao committed:
Suppress 'max-statements' ESLint rule for StateContextProvider.

davemarco reviewed (commented):
Looks better! I went through again (first time was travelling...) and added a few more things

junhaoliao committed:
Rename SidebarContainer to CentralContainer.

junhaoliao committed:
Refactor CentralContainer and extract Sidebar component.

junhaoliao committed:
Reorganize Sidebar component files.

junhaoliao committed:
Rename FileInfoTab to FileInfoTabPanel.

junhaoliao commented on new-log-viewer/src/components/SidebarContainer/PanelTabs/FileInfoTab.tsx:
```diff
@@ -0,0 +1,52 @@
+            tabName={TAB_NAME.FILE_INFO}
+            title={TAB_DISPLAY_NAMES[TAB_NAME.FILE_INFO]}
+        >
+            {0 === fileName.length ?
```
How about
`const isFileOpen = (0 !== fileName.length);`? 

In a future PR where we refactor the loading states, `isFileOpen` can come from the StateContextProvider which can be derived from a different criteria.

junhaoliao committed:
Rename PanelTabs to SidebarTabs.

junhaoliao committed:
Add docs for Sidebar.

junhaoliao committed:
Refactor SidebarTabs to use a constant for tab info.

junhaoliao commented on new-log-viewer/src/components/SidebarContainer/PanelTabs/index.tsx:
```diff
@@ -0,0 +1,93 @@
+                    size={"lg"}
+                >
+                    {[
+                        {tabName: TAB_NAME.FILE_INFO, Icon: InfoOutlinedIcon},
```
Created an array named `TABS_INFO_LIST`. Let me know if I misunderstood your comment.

junhaoliao committed:
Add docs to explain why "sidebar-tab-list-spacing" is needed.

junhaoliao committed:
Split the useEffect hook in ResizeHandle into two.

junhaoliao commented on new-log-viewer/src/components/SidebarContainer/ResizeHandle.tsx:
```diff
@@ -0,0 +1,79 @@
+        onHandleRelease();
+    }, [onHandleRelease]);
+
+    useEffect(() => {
```
Let's split the useEffect hook into separate ones to simplify the relations.

junhaoliao replied:
> The previous one only had the mouse up callback registered when the handle was held

Right

> this one has it active whenever the panel is open

Right. Note that by having the callback always registered, we avoid repetitively registering and deregistering the callback. In both cases, the callback is only invoked whenever there's a "mouseup" event, which is only possible after one has held the mouse as you suggested. :D

I believe the new approach is cleaner and more efficient.

junhaoliao commented on new-log-viewer/src/components/SidebarContainer/index.tsx:
```diff
@@ -0,0 +1,123 @@
+
+    const deactivateTabAndHideResizeHandle = useCallback(() => {
+        setActiveTabName(TAB_NAME.NONE);
+        document.body.style.setProperty("--ylv-panel-resize-handle-width", "0px");
```
Good catch. The `--ylv-panel-resize-handle-width` was added to address an overflow issue in the panel tabs (see `new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/index.css`: `.sidebar-tabs {width: calc(100% - var(--ylv-panel-resize-handle-width));}` but we do not have to dynamically change the resize handle width. You're definitely right that we can just avoid rendering the resize handle when no tab is active.

junhaoliao committed:
Avoid unnecessary dynamic change of the --ylv-panel-resize-handle-width CSS property on tab close.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/PanelTabs/TooltipTab.css:
```diff
@@ -0,0 +1,10 @@
+    width: 48px;
+    height: 48px;
+    padding: 0 !important;
+}
```
`!important` was added to override the JoyUI styles.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/PanelTabs/TooltipTab.css:
```diff
@@ -0,0 +1,10 @@
+
+.sidebar-tab-button-icon {
+    font-size: 32px !important;
+}
```
ditto. `!important` was added to override the JoyUI styles.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/index.css:
```diff
@@ -0,0 +1,13 @@
+    width: 100vw;
+}
+
+.central-container-children-container {
+    width: calc(100vw - var(--ylv-panel-width));
+}
```
Any performance concerns with the proposed approach? Previously I intentionally avoided "flex" usages with the impression that it usually slows down the renderings.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/Sidebar.tsx:
```diff
@@ -0,0 +1,111 @@
+/**
+ *
+ */
+const Sidebar = () => {
```
Should have been addressed in the latest code changes.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/ResizeHandle.css:
```diff
@@ -0,0 +1,16 @@
+.resize-handle:hover {
+    /* stylelint-disable-next-line custom-property-pattern */
+    background-color: var(--joy-palette-primary-solidHoverBg, #0258a8);
+}
```
Those CSS property names are from JoyUI which we cannot easily override. While it's possible to avoid referencing the properties in CSS style sheets by writing CSS-in-JS code to get the color platte, we avoid CSS-in-JS at the moment with the belief that it usually has worse performance than static stylesheets. We can revisit whether we wish to use CSS-in-JS once the Pigment engine is supported in JoyUI: https://mui.com/blog/introducing-pigment-css/

junhaoliao commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/CustomListItem.tsx:
```diff
@@ -0,0 +1,50 @@
+ * @param props.slotProps
+ * @return
+ */
+const CustomListItem = ({content, icon, slotProps, title}: CustomListItemProps) => (
```
I'm not too sure about that. Apparently there are critics on `React.FC` usages: https://github.com/facebook/create-react-app/pull/8177

junhaoliao commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/TooltipTab.tsx:
```diff
@@ -0,0 +1,57 @@
+import {
+    Tab,
+    Tooltip,
+} from "@mui/joy";
+
+import {SvgIconComponent} from "@mui/icons-material";
+
+import {
+    TAB_DISPLAY_NAMES,
+    TAB_NAME,
+} from "../../../../typings/tab";
+
+import "./TooltipTab.css";
```
> use absolute imports to improve readability and maintainability

Will look into that in the future.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/FileInfoTabPanel.tsx:
```diff
@@ -0,0 +1,52 @@
+            {0 === fileName.length ?
+                "No file is open." :
+                <List>
+                    <CustomListItem
+                        content={fileName}
+                        icon={<AbcIcon/>}
+                        slotProps={{content: {sx: {wordBreak: "break-word"}}}}
+                        title={"Name"}/>
+                    <Divider/>
+                    <CustomListItem
+                        content={formatSizeInBytes(originalFileSizeInBytes)}
+                        icon={<StorageIcon/>}
+                        title={"Original Size"}/>
```
The aria attributes are less necessary in this context given we do provide labels (see prop `title`) next to the icons.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/Sidebar/ResizeHandle.tsx:
```diff
@@ -0,0 +1,85 @@
+        isMouseDown,
+    ]);
+
+    return (
+        <div
+            className={"resize-handle"}
+            onMouseDown={handleMouseDown}/>
+    );
+};
```
I appreciate the concerns for accessibility, but providing the panel width to the aria values here are not too meaningful. To properly provide better accessibility, I believe more in-depth discussion are required and are probably out of the scope of this PR.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/Sidebar/index.tsx:
```diff
@@ -0,0 +1,114 @@
+ *
+ * @return the width in pixels as a number.
+ */
+const getPanelWidth = () => parseInt(
+    document.body.style.getPropertyValue("--ylv-panel-width"),
+    10
+);
```
`NaN` is unexpected unless there are coding errors which cause the `--ylv-panel-width` not to be set.

junhaoliao committed:
Change the resize handle width when not hovered.

junhaoliao committed:
Add visual feedback on Sidebar resize handle mouse down.

junhaoliao committed:
Increase panel resize handle width.

junhaoliao commented:
030036eff4c5d38a1bd19eb95c70bb3d52c51b8c Changed the resize handle width when not hovered:

Before:
![image](https://github.com/user-attachments/assets/a9935419-15a5-46cf-8180-0b12816c1ca3)

After:
![image](https://github.com/user-attachments/assets/d4edf9b9-e5a0-47be-a9dd-d5aad96adfa7)

junhaoliao committed:
Rename `TooltipTab` to `TabButton`.

junhaoliao committed:
Merge branch 'main' into sidebar

# Conflicts:
#	new-log-viewer/src/contexts/StateContextProvider.tsx

davemarco reviewed (commented):
I responded to some old comments. I also showed my idea for central container. It's not all the code suggested, but should be most of it. I think it should work. If you are happy with it, you can also move Editor folder into CentralContainer. I think its okay to leave out DropFileContainer

junhaoliao committed:
Introduce a constant for file unloaded check in FileInfoTabPanel.

junhaoliao committed:
Move DropFileContainer and Editor into CentralContainer directly.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/index.css:
```diff
@@ -0,0 +1,13 @@
+
+.central-container {
+    display: grid;
+    grid-template-columns: var(--ylv-panel-width) 1fr;
```
Any benefits with the proposed approach? / any drawbacks of the original approach?

junhaoliao replied:
I see your point. Right, the `central-container-children` class isn't strictly necessary.

junhaoliao committed:
Simplify CentralContainer component documentation - apply code review suggestion.

junhaoliao commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/FileInfoTabPanel.tsx:
```diff
@@ -0,0 +1,53 @@
+            tabName={TAB_NAME.FILE_INFO}
+            title={TAB_DISPLAY_NAMES[TAB_NAME.FILE_INFO]}
+        >
+            {isFileUnloaded ?
+                "No file is open." :
+                <List>
+                    <CustomListItem
+                        content={fileName}
+                        icon={<AbcIcon/>}
+                        slotProps={{content: {sx: {wordBreak: "break-word"}}}}
+                        title={"Name"}/>
+                    <Divider/>
+                    <CustomListItem
+                        content={formatSizeInBytes(originalFileSizeInBytes)}
+                        icon={<StorageIcon/>}
+                        title={"Original Size"}/>
```
Not applicable

junhaoliao committed:
Cache formattedOriginalSize with `useMemo`.

junhaoliao committed:
Reorder params in docs - Apply suggestions from code review

Co-authored-by: davemarco <83603688+davemarco@users.noreply.github.com>

junhaoliao committed:
Remove redundant container "central-container-children" and CSS for CentralContainer layout.

junhaoliao committed:
Merge remote-tracking branch 'junhao/sidebar' into sidebar

junhaoliao committed:
Merge branch 'main' into sidebar

# Conflicts:
#	new-log-viewer/src/contexts/StateContextProvider.tsx

junhaoliao committed:
Increase eslint rule "max-lines" limit to 500.

junhaoliao committed:
Avoid leaving mouseup event listener active even when the resize handle is not being held to prevent `onHandleRelease` being called unintentionally.

davemarco reviewed (dismissed):
Ignore any comments I had on callback, I approve

davemarco committed:
Merge branch 'main' into HEAD

kirkrodrigues commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/FileInfoTabPanel.tsx:
```diff
@@ -0,0 +1,61 @@
+
+
+/**
+ * Displays a panel containing the file name and original size of the selected file.
```
```suggestion
 * Displays a panel containing the file name and on-disk size of the selected file.
```

kirkrodrigues commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/FileInfoTabPanel.tsx:
```diff
@@ -0,0 +1,61 @@
+
+    const isFileUnloaded = 0 === fileName.length;
+    const formattedOriginalSize = useMemo(
+        () => formatSizeInBytes(originalFileSizeInBytes),
```
```suggestion
        () => formatSizeInBytes(originalFileSizeInBytes, false),
```

kirkrodrigues commented on new-log-viewer/src/components/CentralContainer/Sidebar/index.tsx:
```diff
@@ -0,0 +1,105 @@
+import "./index.css";
+
+
+const PANEL_DEFAULT_WIDTH_IN_PIXEL = 360;
+const PANEL_CLIP_THRESHOLD_IN_PIXEL = 250;
```
PIXELS?

kirkrodrigues replied:
Ah no, I mean PIXEL -> PIXELS?

davemarco committed:
Apply suggestions from code review

Co-authored-by: kirkrodrigues <2454684+kirkrodrigues@users.noreply.github.com>

kirkrodrigues commented on new-log-viewer/src/services/LogFileManager/index.ts:
```diff
@@ -91,7 +100,7 @@ class LogFileManager {
         const decoder = await LogFileManager.#initDecoder(fileName, fileData, decoderOptions);
 
-        return new LogFileManager(decoder, fileName, pageSize);
+        return new LogFileManager(decoder, fileName, fileData.length, pageSize);
```
Mm, `sourceFileSize` is a bit abstract.

davemarco committed:
rename onDiskFileSize

davemarco committed:
fix pixels

davemarco committed:
fix lint

kirkrodrigues commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/FileInfoTabPanel.tsx:
```diff
@@ -0,0 +1,61 @@
+    const {fileName, onDiskFileSizeInBytes} = useContext(StateContext);
+
+    const isFileUnloaded = 0 === fileName.length;
+    const formattedOriginalSize = useMemo(
```
formattedOnDiskSize

kirkrodrigues commented on new-log-viewer/src/components/CentralContainer/Sidebar/SidebarTabs/FileInfoTabPanel.tsx:
```diff
@@ -0,0 +1,61 @@
+                    <CustomListItem
+                        content={formattedOriginalSize}
+                        icon={<StorageIcon/>}
+                        title={"Original Size"}/>
```
```suggestion
                        title={"On-disk Size"}/>
```

davemarco committed:
fix original in more places

kirkrodrigues reviewed (approved):
For the PR title, how about:

```
new-log-viewer: Add tabbed Sidebar with resizable panel; Rearrange open and settings buttons.
```

junhaoliao commented on new-log-viewer/src/components/MenuBar/index.css:
```diff
@@ -1,6 +1,28 @@
+    display: flex;
+    justify-content: center;
+    width: 48px;
+    height: var(--ylv-status-bar-height);
```
This should be `--ylv-menu-bar-height` although they both have the same value. Similarly, the `height` in `.menu-bar` should be referencing `--ylv-menu-bar-height` instead as well.

@coderabbitai create a bug issue to track this.
